src/client/Client.cc

```diff
@@ -4288,7 +4288,6 @@ void Client::trim_caps(MetaSession *s, uint64_t max)
   }
   to_trim.clear();
 
-  caps_size = s->caps.size();
```

Comment:
This doesn't look right. The point is if the client still has more caps than the max **after** trimming, it must be because the kernel is keeping the caps pinned. We don't want to invalidate the kernel dcache unnecessarily.

Reply:
understand, thanks.
I used to understand cleaning the kernel before trimming